src/Chess.php

```diff
@@ -1040,6 +1040,8 @@ protected function moveToSAN(Move $move): void
     {
         $cacheKey = json_encode($move).$this->boardHash;
         if (isset($this->sanMoveCache[$cacheKey])) {
+            $move->san = $this->sanMoveCache[$cacheKey];
```

Comment:
so, maybe we need test to confirm this behavior?

Reply:
that's a good point

Reply:
added a test (and squashed)